crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -319,6 +319,7 @@ use std::{cell::UnsafeCell, marker::PhantomData};
 /// # Safety
 ///
 /// component access of `ROQueryFetch<Self>` should be a subset of `QueryFetch<Self>`
+/// and `ROQueryFetch<Self>` should match exactly the same archetypes/tables as `QueryFetch<Self>`
```

Comment:
"should" or "must"?

Reply:
"must" though they feel the same meaning wise in a safety comment, there's no reason to say `/// Safety: you SHOULD do x` since you may aswell not have it be unsafe since noone can rely on `x` xd. will make it say `must` for clarity though